### PR TITLE
Add as_prediction_surv

### PR DESCRIPTION
**Pull request by mllg**
I'm unsure how to convert distr objects back to pass it to PredictionSurv, see FIXME.

### CODE REVIEW

File: R/as_prediction_surv.R [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+#' @rdname as_prediction_surv
+#' @export
+as_prediction_surv.data.frame = function(x, ...) { # nolint
+  mandatory = c("row_ids", "time", "status")
+  optional = c("crank", "lp", "distr")
+  assert_names(names(x), must.include = mandatory)
+  assert_names(names(x), subset.of = c(mandatory, optional))
+
+  # FIXME: distr removed here, unsure how to combine them for PredictionSurv
+  # do.call(c, distr) is incredibly slow
```
Comment (RaphaelS1) [inline R/as_prediction_surv.R:53]: this is a known bottleneck, it affects all of proba; it's in parmetersetcollection (same bottleneck in paradox). doing my best to fix asap. not sure if best to uncomment and have it slow for now or just not have the method until fixed

Comment (mllg) [reply R/as_prediction_surv.R:53]: TBH, I thought I was doing something wrong. Better have it in there and slow than not supporting it.

Comment (RaphaelS1) [reply R/as_prediction_surv.R:53]: Okay can you uncomment then and include distr? Then I'll merge in. Bottleneck will be fixed at the end of the month once new package on cran

**Comment by mllg** [comment]
`do.call(c, distr)` included now, but I'm apparently doing something wrong. I've added a test to illustrate the expected behaviour. Can you have a look?

**Pull request merged**